guides/gitconfig.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Managing Git Configuration
+description: Learn how to manage your Git configuration in Coder.
+---
+
+This guide will show you how to manage Git configuration through the system,
+global, and repo levels.
+
+## How to use the System and Global configurations
+
+For customers seeking organizational (system-level) `.gitconfig` defaults, we recommend
+adding a default configuration into the `/etc/gitconfig` file of each image associated
+to the organization:
+
+```bash
+[url "git@github.com:"]
+    insteadOf = https://github.com/
+```
+
+For users seeking to override the system `.gitconfig` with a global setting of their
+own, we recommend including `.gitconfig` and other personal configuration files in
+the `home/coder` directory. [Learn more about environment personalization.](https://coder.com/docs/environments/personalization#dotfiles-repo)
```

Comment:
We recommend setting a `.gitconfig` in personalize regardless. The fact that it overrides system settings is more pertinent when explaining how to apply gitconfig settings across an organization.

Thus the wording here should start with "We recommend ....". We can re-mention that it overrides system `etc/gitconfig` as an aside, but it's not the leading concept.

Reply:
This makes sense - so I'll clarify with something like:

> While we recommend setting a `.gitconfig` file via the `~/personalize` script, it will override the system-level `etc/gitconfig` default set in the image.

Reply:
I wonder if we can put somewhere in this doc the fact that Coder, by default, will load a `.gitconfig` using your name and email associated with your Coder login. That was a point for me because I was forced to use the dotfiles setup to ensure my `gitconfig` had the proper email address to keep my commits anonymous on GitHub. GitHub allows for private emails, and therefore automatically adding one's email to a gitconfig where they can push to public repositories could expose their email.

We call this out below, but I feel like there's room for highlighting this.

Reply:
Agree - I'm happy to highlight this further. Any thoughts on how we should go about that?

Reply:
Trivial suggestion - maybe the first section of the article can be about our `gitconfig` injection?

Reply:
Good idea - we lead with the default functionality, and then guide the customer through overriding such functionality.